cmds/cmp/cmp.go

```diff
@@ -76,34 +91,26 @@ func main() {
 
 	fnames := flag.Args()
 
-	switch len(fnames) {
-	case 2:
-	case 3:
-		offset[0], err = strconv.ParseInt(fnames[2], 0, 64)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "bad offset1: %s: %v\n", fnames[2], err)
-			return
-		}
-	case 4:
-		offset[0], err = strconv.ParseInt(fnames[2], 0, 64)
-		if err != nil {
-			log.Printf("bad offset1: %s: %v\n", fnames[2], err)
-			return
-		}
-		offset[1], err = strconv.ParseInt(fnames[3], 0, 64)
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "bad offset2: %s: %v\n", fnames[3], err)
-			return
-		}
-	default:
-		log.Fatalf("expected two filenames (and one to two optional offsets), got %d", len(fnames))
+	if nArgs := len(fnames); nArgs < 2 {
+		failf("Too few arguments, expected at least two filenames, got %d arguments", nArgs)
+	} else if nArgs > 4 {
+		failf("Too many arguments, expected exactly two filenames (and one or two optional offsets), got %d arguments", nArgs)
+	}
+
+	cmpUnits := unit.DefaultUnits
+	cmpUnits["kB"] = unit.DefaultUnits["KB"]
+
+	off := unit.MustNewUnit(cmpUnits)
```

Comment:
don't panic in main -- use NewUnit and pass the error to failf

Reply:
Sure, pushed